### PR TITLE
[ch35934] New Notification to UNICEF Focal persons

### PR DESCRIPTION
**Pull request by emaciupe**
[ch35934]

### CODE REVIEW

**Comment by emaciupe** [comment]
requires 
```
./manage.py update_notifications
./manage.py init-celery
```

File: src/etools/applications/partners/tasks.py [hunk 1]
```diff
@@ -272,6 +278,55 @@ def notify_partner_hidden(partner_pk, tenant_name):
...
+        partner_qs = PartnerOrganization.objects\
+            .active()\
+            .annotate(core_value_assessment_expiring=Subquery(core_value_assessment_expiring))\
+            .filter(core_value_assessment_expiring__in=notify_end_dates)\
+            .distinct()
+
+        for partner in partner_qs:
+            pds = Intervention.objects\
+                .prefetch_related('unicef_focal_points')\
+                .filter(agreement__partner=partner)\
```
Comment (robertavram) [inline src/etools/applications/partners/tasks.py:310]: exclude closed and ended interventions
exclude draft interventions with modified more than 1 year ago

File: src/etools/applications/partners/tasks.py [hunk 2]
```diff
@@ -272,6 +278,55 @@ def notify_partner_hidden(partner_pk, tenant_name):
...
+    notify_end_dates = [datetime.timedelta(days=-delta) for delta in _PARTNER_ASSESSMENT_EXPIRING_SOON_DELTAS]
+
+    for country in Country.objects.exclude(name='Global').all():
+        connection.set_tenant(country)
+        logger.info('Starting Partner Assessment Expire for country {}'.format(country.name))
+
+        core_value_assessment_expiring = PartnerOrganization.objects\
+            .filter(pk=OuterRef("pk"))\
+            .annotate(notifs_to_expire=ExpressionWrapper(
+                today - (F('core_values_assessment_date') +
```
Comment (robertavram) [inline src/etools/applications/partners/tasks.py:297]: core_values_assessment_date is one, we also need to add `last_assessment_date` if either of those dates surpass the limit, we need to inform the users which assessment is expiring:
last_assessment_date is for `"HACT Assessment"`

File: src/etools/applications/partners/tasks.py [hunk 3]
```diff
@@ -272,6 +278,55 @@ def notify_partner_hidden(partner_pk, tenant_name):
...
+            .active()\
+            .annotate(core_value_assessment_expiring=Subquery(core_value_assessment_expiring))\
+            .filter(core_value_assessment_expiring__in=notify_end_dates)\
+            .distinct()
+
+        for partner in partner_qs:
+            pds = Intervention.objects\
+                .prefetch_related('unicef_focal_points')\
+                .filter(agreement__partner=partner)\
+                .annotate(unicef_focal_point_emails=ArrayAgg('unicef_focal_points__email'))
```
Comment (robertavram) [inline src/etools/applications/partners/tasks.py:311]: exclude users that don't have the profile_country set to the current country.

**Review by robertavram — CHANGES_REQUESTED** [review]

**Pull request merged**